code/modules/clothing/under/accessories/badges.dm

```diff
@@ -7,7 +7,8 @@
 /obj/item/clothing/accessory/badge
 	name = "badge"
 	desc = "A leather-backed badge, with gold trimmings."
-	icon_state = "detectivebadge"
+	icon = 'icons/clothing/mask/detectivebadge.dmi'
```

Comment:
This is using an `accessory_icons` entry instead of the icon in the file, review